src/main/java/seedu/duke/ModuleList.java

```diff
@@ -0,0 +1,175 @@
+
+    /**
+     * Function to add a module to the moduleList if it does not already exist
+     * @param mod The module to be added to the list
+     */
+    public void add(Module mod) {
+        boolean isRepeat = checkRepetition(mod);
+        if (isRepeat) {
+            UI.repetitionMessage(mod.getCourse());
+        } else {
+            int before = modules.size();
+            modules.add(mod);
+            UI.addMessage(mod.getCourse(), mod.getSemesterTaken(), mod.getGrade());
+            int after = modules.size();
+
+            try {
+                assert before + 1 == after : "module list size should be increased by 1 after add";
+            } catch (AssertionError e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * function to check if a given module already exists in the moduleList
+     * @param mod the module that needs to be searched for in the moduleList
+     * @return true if the module already exists. False if it does not. Format: boolean
+     */
+    public boolean checkRepetition(Module mod) {
+        for (Module m : modules) {
+            if (m.getCourse().matches(mod.getCourse())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * function to delete a module from the moduleList. If the module is not found, an appropriate message is displayed
+     * @param modCode the module which has to be deleted from the moduleList.
+     */
+    public void delete(String modCode) {
+        int indexCounter = 0;
+        boolean isFound = false;
+        for (Module mod : modules) {
+            if (mod.getCourse().matches(modCode)) {
+                int before = modules.size();
+                modules.remove(indexCounter);
+                UI.deleteMessage(modCode);
+                isFound = true;
+                int after = modules.size();
+                try {
+                    assert before == after + 1 : "module list size should be decreased by 1 after delete";
+                } catch (AssertionError e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            }
+            indexCounter += 1;
+        }
+        if (!isFound) {
+            UI.notFoundDeletionMessage(modCode);
+        }
+    }
+
+    /**
+     * Function to find a list of all modules taken in a particular semester
+     * The function then calls the printResponse() method to print the appropriate response message
+     * @param semester The semester for which the modules need to be printed
+     */
+    public void view(String semester) {
+        ArrayList<Module> matchingModules = new ArrayList<>();
+        viewCount = 0;
+        for (Module mod : modules) {
+            if (mod.getSemesterTaken().matches(semester)) {
+                matchingModules.add(mod);
+                viewCount++;  // increments the count for this particular semester (For JUnit Purpose)
```

Comment:
The comment is a little confusing. Consider removing it or rephrase into explain the rationale behind the variable. For example, "added view count for each semster for easier testing".